### PR TITLE
docs: fold the site from 63 pages to 37, every old URL redirected

### PR DESCRIPTION
**Pull request by vboussot**
Stacked on #178 (the audit branch): retarget to `main` once that one merges, before any branch deletion.

The site goes from 63 source files and 47 sidebar entries to 37 files and 30 entries, merged by the reader's question rather than by the old section: the reflection engine, the four-files table and the shared `Dataset` conventions become one Configuration page; the streaming concept and the benchmark protocol join the large-images guide; model-graph naming and the YAML builder join the models catalogue; the extension-point contracts join the custom-models guide; the core and apps Python APIs and their autodoc signatures become one page; ecosystem joins apps, environment joins the CLI, augmentations join transforms, schedulers join losses, the Docker README joins installation, the two thin example wrappers join the examples overview.

Every old URL still resolves: `conf.py` writes a meta-refresh stub at each old path on `build-finished` (26 entries, anchors checked against the built ids); README links point at the new homes; `llms.txt` follows. Strict build: 0 warnings before and after.

Two claims corrected on the way: an undecorated local model is nested under `Trainer.Model.<Class>` like a decorated one, and `outputs_criterions` lives inside that subtree, as the shipped configs have it.

## Summary by CodeRabbit

- **Documentation**
  - Reorganized navigation and refreshed links across configuration, Python APIs, apps, models, transforms, schedulers, and large-image guidance.
  - Added expanded guidance for workflow commands, custom extensions, model construction, patch streaming, augmentations, and app execution.
  - Reworked the quickstart around a CPU-only synthetic segmentation example.
  - Expanded API and CLI references and added redirects for relocated pages.
  - Removed or consolidated legacy concepts, workflow, benchmark, ecosystem, Docker, environment, and API reference pages.

### CODE REVIEW

File: docs/source/examples/visual-gallery.md [hunk 1]
```diff
@@ -211,6 +211,6 @@ plane at a time.
 ## Next steps
 
 - {doc}`../reference/components/transforms`: every transform and argument
-- {doc}`../reference/components/augmentations`: every augmentation and lifecycle detail
+- {doc}`../reference/components/transforms`: every augmentation and lifecycle detail
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/examples/visual-gallery.md]: Fixed in aedfc7f6: the bullet links to the Augmentations section.

File: docs/source/usage/custom-models.md [hunk 2]
```diff
@@ -258,6 +249,242 @@ groups_dest:
...
+`patch_locality` only once you can state which kind is honest for it.
+
+Two traps:
+
+- **A random per-voxel augmentation is not `POINTWISE`.** The kind is about the
+  voxel's position, not the arithmetic's shape. A field drawn per call is a
+  different field on every call, so overlapping patches sample unrelated fields
+  and the overlap blend suppresses the variance the augmentation exists to add.
+  The built-in `Noise` declares `WHOLE_VOLUME` for this reason. Declared
+  `POINTWISE`, two reads of the same patch return different values.
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/usage/custom-models.md]: Fixed in aedfc7f6: Noise is POINTWISE and the bullet says so.

File: docs/source/usage/large-images.md [hunk 3]
```diff
@@ -163,11 +180,353 @@ designed to stay correct. Measure peak RSS on a representative case, and hold th
 model, patch size, overlap, TTA, batch size, workers and hardware fixed when you
 compare.
 
+## Patch streaming
+
+**KonfAI works out of core.** A case never has to fit in RAM: each patch's source
+region is read straight from the file, and the result is written slab by slab as
+it completes. Neither the input nor the output is ever held whole. A 16 GiB
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/usage/large-images.md]: Fixed in aedfc7f6: the claim is limited to streamable chains and names the whole-volume path.

File: docs/source/usage/custom-models.md [hunk 4]
```diff
@@ -60,58 +58,50 @@ instantiate eagerly.
...
+class name when the class has no `_key`, so an undecorated `UNetpp5` reads
+from `Trainer.Model.UNetpp5`, exactly where `@config()` would put it. For a
+model a decorator only *renames* that subtree:
 
 ```yaml
 Trainer:
   Model:
     classpath: Model:UNetpp5
+    UNetpp5:
+      ...                 # the model's own arguments, decorated or not
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/usage/custom-models.md:83]: Fixed in 1bd9ddd1: outputs_criterions nested under UNetpp5.

File: docs/source/usage/custom-models.md [hunk 5]
```diff
@@ -379,6 +605,141 @@ outputs_criterions:
...
+    """Mean of the members left after dropping each voxel's min and max."""
+
+    voxel_local = True     # every output voxel reads only the SAME voxel of each member
+    incremental = False    # __call__ needs all members at once
+    working_multiple = 4.0 # the stacked float copy plus the reduction buffers
+
+    def __call__(self, tensors: list[torch.Tensor]) -> torch.Tensor:
+        stack = torch.stack([tensor.float() for tensor in tensors])
+        trimmed = stack.sum(dim=0) - stack.amax(dim=0) - stack.amin(dim=0)
+        return trimmed / (len(tensors) - 2)
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/usage/custom-models.md:717]: Fixed in 1bd9ddd1: TrimmedMean refuses fewer than three members.
